bootloader/piboot.go

```diff
@@ -0,0 +1,395 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
```

Comment:
this is so different from other bootloaders that I think we need a piboot.md or piboot.txt in the directory that explains how it is supposed to work, the code is fairly non-obvious as is